aws-curl

```diff
@@ -499,6 +507,22 @@ fi
 if [ "$EC2_CREDS" = 1 ]; then
     ec2_import_creds
 fi
+get_cred_value() {
+    echo "$1" | grep "$2 =" | cut -d ' ' -f 3-
+}
+
+if [ -n "$AWS_PROFILE" ] && [ -z "$AWS_ACCESS_KEY_ID" ]; then
+  # this can work with AWS SSO based connections
+  block=$(sed -n '/\['$AWS_PROFILE'/,/^$/p' ~/.aws/credentials)
+  AWS_ACCESS_KEY_ID=$(get_cred_value "$block" aws_access_key_id)
```

Comment:
This code doesn't handle situation when profile doesn't have values for parameters. It blindly overwrites context with what it is profile, but if profile has no values and values were passed in script in alternative way, then these values will be zeroed.